OpenRA.Mods.Common/Activities/SpriteHarvesterDockSequence.cs

```diff
@@ -15,24 +15,34 @@ namespace OpenRA.Mods.Common.Activities
 {
 	public class SpriteHarvesterDockSequence : HarvesterDockSequence
 	{
-		readonly RenderUnit ru;
+		readonly WithSpriteBody wsb;
+		readonly WithDockingAnimation wda;
 
 		public SpriteHarvesterDockSequence(Actor self, Actor refinery, int dockAngle, bool isDragRequired, WVec dragOffset, int dragLength)
 			: base(self, refinery, dockAngle, isDragRequired, dragOffset, dragLength)
 		{
-			ru = self.Trait<RenderUnit>();
+			wsb = self.Trait<WithSpriteBody>();
+			wda = self.Trait<WithDockingAnimation>();
 		}
 
 		public override Activity OnStateDock(Actor self)
 		{
-			ru.PlayCustomAnimation(self, "dock", () => ru.PlayCustomAnimationRepeating(self, "dock-loop"));
+			var wha = self.TraitOrDefault<WithHarvestAnimation>();
+			if (wha != null)
+				wha.IsModifying = true;
```

Comment:
Ugh that is ... not nice.
Can you add a comment to explain this workaround please.